test/_utils/test_scenario.py

```diff
@@ -47,7 +47,10 @@ def test_multivariate_simulation_2D(
         seed=seed,
     )
 
-    signal_noise_ratio_hat = np.linalg.norm(y - noise) / np.linalg.norm(noise)
+    if signal_noise_ratio != np.inf:
```

Comment:
Ne we don't want to have np.inf all over the place. We need to regularize the noise norm.

Reply:
What do you mean by regularize the noise norm?

Reply:
```suggestion
signal_noise_ratio_hat = np.linalg.norm(y - noise) / max(np.linalg.norm(noise), 1e-6)
```
I think something like that should solve the division by 0. It will however, not work for the `assert_almost_equal` test

Reply:
> The inf is the case where there is no noise in the signal which seems for me a good value.

In engineering, the 0-noise case does not exist. Having an infinite SNR does not make sense.  An infinite value is not a a large value, it simply means that there is a numerical issue. I don't agree with this test.

Reply:
Yes, but the default value should not be `None` in that case. It could  be 0.

Reply:
I prefer a default value of 1.

Reply:
OK